sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobAsyncClient.java

```diff
@@ -278,23 +280,28 @@
      *
      * <p><strong>Code Samples</strong></p>
      *
-     * {@codesnippet com.azure.storage.blob.specialized.cryptography.EncryptedBlobAsyncClient.uploadWithResponse#Flux-BlobParallelUploadOptions}
+     * {@codesnippet com.azure.storage.blob.specialized.cryptography.EncryptedBlobAsyncClient.uploadWithResponse#BlobParallelUploadOptions}
      *
-     * @param data The data to write to the blob. Unlike other upload methods, this method does not require that the
      * {@code Flux} be replayable. In other words, it does not have to support multiple subscribers and is not expected
      * to produce the same values across subscriptions.
      * @param options {@link BlobParallelUploadOptions}
      * @return A reactive response containing the information of the uploaded block blob.
      */
     @Override
-    public Mono<Response<BlockBlobItem>> uploadWithResponse(Flux<ByteBuffer> data, BlobParallelUploadOptions options) {
+    public Mono<Response<BlockBlobItem>> uploadWithResponse(BlobParallelUploadOptions options) {
         try {
-            BlobParallelUploadOptions finalOptions = options == null ? new BlobParallelUploadOptions() : options;
-            final Map<String, String> metadataFinal = finalOptions.getMetadata() == null
-                ? new HashMap<>() : finalOptions.getMetadata();
-            finalOptions.setMetadata(metadataFinal);
+            Objects.requireNonNull(options);
+            final Map<String, String> metadataFinal = options.getMetadata() == null
+                ? new HashMap<>() : options.getMetadata();
+            options.setMetadata(metadataFinal);
```

Comment:
Some say that mutating input parameters isn't good practice. 

https://softwareengineering.stackexchange.com/questions/245767/is-modifying-an-incoming-parameter-an-antipattern

looks like we do a copy few lines below anyway.

Reply:
I don't have to set it on the options. I can use a local variable if that's better?

Reply:
sounds good,

Reply:
btw. Collections.emtpyMap() might be more optimal.